sdk/batch/microsoft-azure-batch/src/main/java/com/microsoft/azure/batch/JobOperations.java

```diff
@@ -621,7 +621,7 @@ public TaskCounts getTaskCounts(String jobId, Iterable<BatchClientBehavior> addi
         BehaviorManager bhMgr = new BehaviorManager(this.customBehaviors(), additionalBehaviors);
         bhMgr.applyRequestBehaviors(options);
 
-        return this.parentBatchClient.protocolLayer().jobs().getTaskCounts(jobId, options);
+        return this.parentBatchClient.protocolLayer().jobs().getTaskCounts(jobId, options).taskCounts();
```

Comment:
This should likely either change return types to the TaskCountsResult model or another method should be added exposing taskSlotCounts as well